### PR TITLE
refactor!: info button refactor

### PR DESCRIPTION
**Pull request by joe-sayer**
## 🖼 Context

This PR is necessary to expose info to push consistency for the info button implementation and avoid code duplication.

## 🚀 Changes

 
- BREAKING CHANGE: Make Summary/Info aria-label mandatory
- Expose info button
- Update docs

## ✅ Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [x] You've updated the documentation if necessary

### CODE REVIEW

File: packages/docs/pages/components/content/Info.mdx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Info } from '@transferwise/components';
+
+import { LiveEditorBlock, GeneratePropsTable } from '../../../utils';
+import code from '../../../liveEditorCode/info.code';
+
+Opens a Popover or Modal based on presentation prop.
```
Comment (dozortsev) [inline packages/docs/pages/components/content/Info.mdx:6]: Could you please add more meaningful and useful documentation? What this component for? when it's recommended to use it? when it's recommended to use Popover presentation and when Modal one? etc

File: packages/components/src/info/Info.js [hunk 2]
```diff
@@ -21,6 +21,12 @@ const Info = (props) => {
     children: <HelpCircleIcon size={isSmall ? 16 : 24} />,
```
[the remark this answers is not in the document]
Comment (andreapiras) [reply packages/components/src/info/Info.js:21]: I think this is quite a specific pattern that we should enforce. I don't think people should be able to decide which icon to use. If in future we'll decide to change this and extend it we can add a prop to allow user to chose from a defined set of icons.

File: packages/components/src/info/Info.js [hunk 3]
```diff
@@ -50,11 +61,13 @@ Info.Presentation = {
...
 
 Info.propTypes = {
-  'aria-label': Types.string,
+  'aria-label': Types.string.isRequired,
   /** Extra classes applied to Info */
   className: Types.string,
   /** Content displayed inside a Popover a Modal */
   content: Types.node,
+  /** Function called on open */
+  onOpen: Types.func,
```
[the remark this answers is not in the document]
Comment (andreapiras) [reply packages/components/src/info/Info.js:70]: why not?

File: packages/docs/pages/components/content/Info.mdx (hunk 1 above)
Comment (andreapiras) [reply packages/docs/pages/components/content/Info.mdx:6]: I think this should be documented in the docs

File: packages/components/src/info/Info.js [hunk 4]
```diff
@@ -30,11 +36,16 @@ const Info = (props) => {
...
-          <button type="button" onClick={() => setOpen(true)} {...buttonProps} />
+          <button type="button" onClick={() => setOpen(!open)} {...buttonProps} />
           <Modal body={content} onClose={() => setOpen(false)} open={open} title={title} />
         </>
       ) : (
-        <Popover content={content} preferredPlacement={Popover.Placement.BOTTOM_LEFT} title={title}>
+        <Popover
+          content={content}
+          preferredPlacement={Popover.Placement.BOTTOM_LEFT}
+          onOpen={setOpen}
```
[the remark this answers is not in the document]
Comment (andreapiras) [reply packages/components/src/info/Info.js:46]: We can and won't make any difference but I'll have to replicate the logic to check whether onOpen is defined and also I like the idea that onOpen and the state of Info are in sync

[the remark this answers is not in the document]
Comment (andreapiras) [reply packages/components/src/info/Info.js]: we postponed this after Popover gets released

File: packages/docs/pages/components/content/Info.mdx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+<div>
+  <LiveEditorBlock code={code} scope={{ Info }} />
+  <GeneratePropsTable componentName="Info" />
+</div>
+
+export const meta = {
+  name: 'Info',
+  badge: {
+    type: 'new',
+    expiryDate: '31-03-2021',
```
Comment (dozortsev) [inline packages/docs/pages/components/content/Info.mdx:17]:
```suggestion
    expiryDate: '2021-03-31',
```

File: packages/components/src/info/Info.story.js [hunk 6]
```diff
@@ -19,11 +20,12 @@ export const basic = () => {
 
   return (
     <Info
+      aria-label="Click here for more details"
       content={content}
+      onOpen={action('open')}
```
Comment (helendurranttw) [inline packages/components/src/info/Info.story.js:25]: Not sure why onOpen was removed? But if we're going to remove it it should be removed here too

Comment (andreapiras) [inline packages/components/src/info/Info.story.js:25]:
```suggestion
      onOpen={action('open')}
```

File: packages/components/src/info/Info.story.js [hunk 7]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import { text, select } from '@storybook/addon-knobs';
+import { action } from '@storybook/addon-actions';
```
Comment (andreapiras) [inline packages/components/src/info/Info.story.js:3]:
```suggestion
```

File: packages/components/src/info/Info.story.js (hunk 6 above)
Comment (andreapiras) [reply packages/components/src/info/Info.story.js]: We postponed this change after Popover release

**Pull request merged**